src/Adafruit_SPIFlashBase.h

```diff
@@ -75,6 +75,8 @@ class Adafruit_SPIFlashBase {
   int _ind_pin;
   bool _ind_active;
 
+  void write_status_register(uint8_t *);
```

Comment:
the prototype should be 

```C
void write_status_register(uint8_t const full_status[2]);
```

Reply:
Done in https://github.com/adafruit/Adafruit_SPIFlash/pull/109/commits/c38a7ca3e65f328310936c12f62d860ed5c164cd

This PR is intentionally designed to create of no or least harm possible:
- it uses current API only
    * setReadCommand() on an intermediate level and
    * supports_qspi=false on a 'user level' (in flash device descriptor)
- default behavior remains the same as before - QSPI will operate at quad data rate unless user have not supplied supports_qspi=false argument intentionally.

It is out of the scopes of this PR to alter current API of the library.
I suppose that any changes in the API are necessary only when there are high demands of users to use extended features.
I doubt that this 1-bit access feature is currently so important that we need to change internal and external API of the library.

Reply:
thanks for confirmation, you don't have to alter the current API. IMHO, the flash device table is flash description, if the flash itself support quad mode, we should just leave it as it is. Instead We could update the actual transport layer. E.g 1bit mode or 2bit mode can be activated when declaring an object with `IO1-IO3 = -1`

https://github.com/adafruit/Adafruit_SPIFlash/blob/master/src/qspi/Adafruit_FlashTransport_QSPI_NRF.cpp#L35

the rest of the code can still apply, and it is transparent to existing user. That also allow using existing flash dev database and the n-bit mode is dictated by the user sketch only. Let me know if this makes sense to you.

Reply:
will be happy to see (and use) your own PR that will supersede this one

Reply:
I would like to, but I don't have motivation (and time) unless adafruit make board that need this. Meanwhile, we can leave this PR pending as referenace.